x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/rules_feature_tour_context.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, useContext, useEffect, useMemo, FC } from 'react';
+
+import { noop } from 'lodash';
+import {
+  useEuiTour,
+  EuiStatelessTourStep,
+  EuiSpacer,
+  EuiButton,
+  EuiTourStepProps,
+} from '@elastic/eui';
+import { invariant } from '../../../../../../common/utils/invariant';
+import { useKibana } from '../../../../../common/lib/kibana';
+
+import * as i18n from '../translations';
+
+export interface RulesFeatureTourContextType {
+  steps: {
+    inMemoryTableStepProps: EuiTourStepProps;
+    bulkActionsStepProps: EuiTourStepProps;
+  };
+  goToNextStep: () => void;
+  finishTour: () => void;
+}
+
+const STORAGE_KEY = 'securitySolution.rulesFeatureTour';
+const TOUR_POPOVER_WIDTH = 360;
+
+const featuresTourSteps: EuiStatelessTourStep[] = [
+  {
+    step: 1,
+    title: i18n.FEATURE_TOUR_IN_MEMORY_TABLE_STEP_TITLE,
+    content: <></>,
+    stepsTotal: 2,
+    children: <></>,
+    onFinish: noop,
+    maxWidth: TOUR_POPOVER_WIDTH,
+  },
+  {
+    step: 2,
+    title: i18n.FEATURE_TOUR_BULK_ACTIONS_STEP_TITLE,
+    content: <p>{i18n.FEATURE_TOUR_BULK_ACTIONS_STEP}</p>,
+    stepsTotal: 2,
+    children: <></>,
+    onFinish: noop,
+    anchorPosition: 'rightUp',
+    maxWidth: TOUR_POPOVER_WIDTH,
+  },
+];
+
+const tourConfig = {
+  currentTourStep: 1,
+  isTourActive: true,
+  tourPopoverWidth: TOUR_POPOVER_WIDTH,
+  tourSubtitle: i18n.FEATURE_TOUR_TITLE,
+};
```

Comment:
> I don't think it can be issue. For each release we want to prepare another tour for different features.

Yes, that's why I was saying "in general". In this particular case, it will work if we include the version into the key and won't ever read it in 8.2+. In other cases, I'd try to avoid such patterns of data persistence in local storage.

> I think, having current step there is a nice thing and we should do it, to prevent showing to user multiple time same step.

Ok, so do we change the persistent state to having only two fields (tour visible + current step) or leave everything as is?

Reply:
> Ok, so do we change the persistent state to having only two fields (tour visible + current step) or leave everything as is?

I've already changed to 2 fields. Let's leave it then as 2. But, in retrospective, i think even 4 fields was fine(as code was simpler). But that's a tradeoff, I'm happy with